src_c/transform.c

```diff
@@ -2966,6 +2966,152 @@ surf_average_color(PyObject *self, PyObject *args, PyObject *kwargs)
+                    sum += *(srcpx + src->pitch * y + nb * (x + radius + 1) +
+                             color);
+                }
+            }
+        }
+
+        for (x = 0; x < h; x++) {
+            sum = 0;
+            for (i = 0; i <= radius; i++) {
+                sum += *(dstpx + dst->pitch * i + nb * x + color);
+            }
+            for (y = 0; y < h; y++) {
+                vbuf[y] = (Uint8)MIN(sum / (radius * 2 + 1), 255);
+
+                // update sum
+                if (y - radius >= 0) {
+                    sum -=
+                        *(dstpx + dst->pitch * (y - radius) + nb * x + color);
+                }
+                if (y + radius + 1 < h) {
+                    sum += *(dstpx + dst->pitch * (y + radius + 1) + nb * x +
+                             color);
+                }
+            }
+
+            for (y = 0; y < h; y++) {
+                // write back from buffer
+                *(dstpx + dst->pitch * y + nb * x + color) = vbuf[y];
+            }
+        }
+    }
+    free(vbuf);
+}
+
+static SDL_Surface *
+blur(pgSurfaceObject *srcobj, pgSurfaceObject *dstobj, int radius)
+{
+    SDL_Surface *src = NULL;
+    SDL_Surface *retsurf = NULL;
+
+    if (radius < 0) {
+        return RAISE(PyExc_ValueError, "Radius less than zero.");
+    }
+
+    src = pgSurface_AsSurface(srcobj);
+
+    if (!dstobj) {
+        retsurf = newsurf_fromsurf(src, src->w, src->h);
+        if (!retsurf)
+            return NULL;
+    }
+    else {
+        retsurf = pgSurface_AsSurface(dstobj);
+    }
+
+    if ((retsurf->w) != (src->w) || (retsurf->h) != (src->h)) {
+        return RAISE(PyExc_ValueError,
+                     "Destination surface not the same size.");
+    }
+
+    if (src->format->BytesPerPixel != retsurf->format->BytesPerPixel) {
+        return (SDL_Surface *)(RAISE(
+            PyExc_ValueError,
+            "Source and destination surfaces need the same format."));
+    }
+
+    SDL_LockSurface(retsurf);
+    pgSurface_Lock(srcobj);
+
+    Py_BEGIN_ALLOW_THREADS;
+
+    box_blur(src, retsurf, radius);
+
+    Py_END_ALLOW_THREADS;
+
+    pgSurface_Unlock(srcobj);
+    SDL_UnlockSurface(retsurf);
+
+    return retsurf;
+}
+
+static PyObject *
+surf_blur(PyObject *self, PyObject *args, PyObject *kwargs)
+{
+    pgSurfaceObject *dst_surf_obj = NULL;
+    pgSurfaceObject *src_surf_obj;
+    SDL_Surface *new_surf = NULL;
+
+    int radius;
+
+    static char *kwlist[] = {"surface", "radius", "dest_surface", 0};
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwargs, "O!i|O!", kwlist,
+                                     &pgSurface_Type, &src_surf_obj, &radius,
+                                     &pgSurface_Type, &dst_surf_obj))
+        return NULL;
+
+    new_surf = blur(src_surf_obj, dst_surf_obj, radius);
+    if (!new_surf) {
+        return NULL;
+    }
+
+    if (dst_surf_obj) {
+        Py_INCREF(dst_surf_obj);
+        return (PyObject *)dst_surf_obj;
+    }
+    else {
```

Comment:
Code quality nitpick: Could you remove the else statement and put the return new surface line at the end of the function?